### PR TITLE
Add omicron levels coloring

### PR DESCRIPTION
**Pull request by jameshadfield**
For background please see "Variant report 2022-10-17" section of https://github.com/neherlab/SARS-CoV-2_variant-reports/blob/main/reports/variant_report_latest_draft.md for details of lineage classification. This commit is very much a first pass to see if (a) it works robustly on large datasets and (b) there is interest in including it (there is from me!)

Note that the mutations are relative to BA.2 so we should really use this for omicron-only builds, which are being developed elsewhere. This is especially relevant for S:493 where we consider a reversion to increase the level, meaning strains basal to BA.2 will be elevated inappropriately.

Trial builds via https://github.com/nextstrain/ncov/actions/runs/3278003787. 
nextstrain URLs: https://nextstrain.org/staging/ncov/gisaid/trial/feat-levels/REGION_NAME
e.g.  https://nextstrain.org/staging/ncov/gisaid/trial/feat-levels/global/2m
You can attach to this AWS job via:
`nextstrain build --aws-batch --attach 08b4c304-03ea-49f6-af22-f60102f14896 /home/runner/work/ncov/ncov`
View this job in the AWS console via
    https://console.aws.amazon.com/batch/home?region=us-east-1#jobs/detail/08b4c304-03ea-49f6-af22-f60102f14896

cc @corneliusroemer

### CODE REVIEW

File: scripts/assign_levels.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import json
+import argparse
+
+# https://github.com/neherlab/SARS-CoV-2_variant-reports/blob/main/reports/variant_report_latest_draft.md
+key_aa_positions = (
+    # [0] is the basal codon, [1] is the 1-based spike aa position, [2] is a tuple of codons which result in a level elevation
+    ("R", 346, ("T", "S")),
+    ("K", 356, ("T")),
+    ("K", 444, ("T", "R")),
+    ("V", 445, ("P")),
```
Comment (corneliusroemer) [inline scripts/assign_levels.py:11]:
```suggestion
    ("V", 445, ("A","P")),
```

File: scripts/assign_levels.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from Bio import SeqIO
+import json
+import argparse
+
+# https://github.com/neherlab/SARS-CoV-2_variant-reports/blob/main/reports/variant_report_latest_draft.md
+key_aa_positions = (
+    # [0] is the basal codon, [1] is the 1-based spike aa position, [2] is a tuple of codons which result in a level elevation
+    ("R", 346, ("T", "S")),
+    ("K", 356, ("T")),
+    ("K", 444, ("T", "R")),
```
Comment (corneliusroemer) [inline scripts/assign_levels.py:10]:
```suggestion
    ("K", 444, ("T", "R","N","M")),
```

File: scripts/assign_levels.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    ("V", 445, ("P")),
+    ("G", 446, ("S")),
```
Comment (corneliusroemer) [inline scripts/assign_levels.py:12]:
```suggestion
    ("G", 446, ("S","D")),
```

File: scripts/assign_levels.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    ("N", 450, ("D")),
+    ("L", 452, ("R")),
```
Comment (corneliusroemer) [inline scripts/assign_levels.py:14]:
```suggestion
    ("L", 452, ("R","Q","M")),
```

File: scripts/assign_levels.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    ("N", 460, ("K")),
+    ("F", 486, ("S", "V", "L", "P")),
+    ("F", 490, ("S")),
```
Comment (corneliusroemer) [inline scripts/assign_levels.py:17]:
```suggestion
    ("F", 490, ("S","V","L","P")),
```

**Review by corneliusroemer — CHANGES_REQUESTED** [review]
Really good

The level definitions are somewhat arbitrary at the boundaries - they are informed by what we commonly see - could be subject to change.

**Comment by corneliusroemer** [comment]
Excellent stuff @jameshadfield.

1. It'd be great if we could make non-21L and descendants grey, as opposed to 0
2. Maybe call it `RBD level 7` etc.

It's great, definitely _already_ useful as is, could just merge and improve over time. Maybe greying out non-21L descendants is the only important change.

**Comment by jameshadfield** [comment]
Thanks @corneliusroemer. I'll move the definitions out of a script into a config (easy), change the name to `RBD level X` (easy) and see if I can make non-21L genomes grey (should be easy?). Watch this space...

**Comment by jameshadfield** [comment]
Applied @corneliusroemer's changes to the definitions and generally made this PR into something that is maintainable going forward. Changes to the level definitions are made by modifying `defaults/rbd_levels.yaml` rather than the script itself. 

But I see that CI has failed because "ERROR: no node found with a clade_annotation of 21L (Omicron)". We want to allow this to work on all trees, even if they don't have any omicron sequences, so I'll update this now.

**Comment by jameshadfield** [comment]
Test builds (GenBank this time) running now. I'll merge this as long as it completes succsessfully:

--> Trial name is: feat-levels-mk2

--> When completed, the following will be available:
build files: s3://nextstrain-staging/files/ncov/open/trial/feat-levels-mk2/
nextstrain URLs: https://nextstrain.org/staging/ncov/open/trial/feat-levels-mk2/REGION_NAME

--> You can attach to this AWS job via:
nextstrain build --aws-batch --attach 4ef45461-1577-4d7c-ae7b-6c4373b37a0d /home/runner/work/ncov/ncov

--> View this job in the AWS console via
    https://console.aws.amazon.com/batch/home?region=us-east-1#jobs/detail/4ef45461-1577-4d7c-ae7b-6c4373b37a0d

**Comment by jameshadfield** [comment]
That trial build failed, here's another one...

--> Trial name is: feat-levels-mk3

--> When completed, the following will be available:
build files: s3://nextstrain-staging/files/ncov/open/trial/feat-levels-mk3/
nextstrain URLs: https://nextstrain.org/staging/ncov/open/trial/feat-levels-mk3/REGION_NAME

--> You can attach to this AWS job via:
nextstrain build --aws-batch --attach 9ec158cd-dda8-4e42-b6dd-89b8327f93f4 /home/runner/work/ncov/ncov

--> View this job in the AWS console via
    https://console.aws.amazon.com/batch/home?region=us-east-1#jobs/detail/9ec158cd-dda8-4e42-b6dd-89b8327f93f4

**Comment by jameshadfield** [comment]
[Trial build looks good](https://next.nextstrain.org/staging/ncov/open/trial/feat-levels-mk3/global/2m?c=rbd_level)

image

**Comment by jameshadfield** [comment]
[the following discussion from slack]
* I've changed the title from `RBD levels` to `RBD Levels`
* The frequencies panel is not colored appropriately due to an ordinal scale. This is a simple fix on Auspice's side (https://github.com/nextstrain/auspice/issues/843) which I'll get to ASAP

**Pull request merged**